src/components/TransactionBlock/AssetSwapTransactionBlock.tsx

```diff
@@ -151,6 +157,29 @@ const AssetSwapTransactionBlock = ({
     [sdk, selectedFromAsset, selectedToAsset, amount, selectedNetwork, accountAddress]
   );
 
+  const getGasSwapUsdValue = async (offer: ExchangeOffer) => {
+    if (!selectedNetwork?.chainId) return;
+
+    const sdkByChain = getSdkForChainId(selectedNetwork?.chainId);
+
+    if (sdkByChain && selectedFromAsset && selectedAccountType === AccountTypes.Contract) {
+      await sdkByChain.computeContractAccount();
+
+      sdkByChain.clearGatewayBatch();
+
+      await Promise.all(
+        offer.transactions.map((transaction) => sdkByChain.batchExecuteAccountTransaction(transaction))
```

Comment:
this would fail if smart contract account is not  deployed so u wanna add deployment tx tto batch if state is not deployed (can check SDK state)